base/sparse/sparsematrix.jl

```diff
@@ -120,7 +120,103 @@ column. In conjunction with [`nonzeros`](@ref) and
 """
 nzrange(S::SparseMatrixCSC, col::Integer) = S.colptr[col]:(S.colptr[col+1]-1)
 
+
+module SparseArrayValidity
+    @enum(_SparseArrayValidity,
+        VALID,
+        COLPTR_LENGTH,
+        COLPTR_FIRST_VAL,
+        NZVAL_LENGTH,
+        COLPTR_SORTED,
+        ROWVAL_LENGTH,
+        ROWVAL_RANGE,
+        ROWVAL_SORTED_COLUMN,
+    )
+end
+
+
+"""
+    checkvalid(Bool, S::SparseMatrixCSC; full = true) -> Bool
```

Comment:
this isn't right as a signature, it would locally overwrite Bool as a variable name